Sources/SwiftRepo/Repository/Protocols/Repository/QueryRepository.swift

```diff
@@ -37,7 +37,7 @@ public protocol QueryRepository<QueryId, Variables, Key, Value>: HasValueResult
         errorIntent: ErrorIntent,
         queryStrategy: QueryStrategy?,
         willGet: @escaping Query.WillGet
-    ) async
+    ) async throws
```

Comment:
@wtmoose I made this throwing, but I think we had agreed that errors caught in this function (would come from throwing store related APIs) should be propagated through the publisher as normal, right? I need to fix this